tests/testImageMapReduce.py

```diff
@@ -0,0 +1,243 @@
+
+    def __init__(self, *args, **kwargs):
+        """! Create the image gridding subTask
+        @param *args arguments to be passed to lsst.pipe.base.task.Task.__init__
+        @param **kwargs keyword arguments to be passed to lsst.pipe.base.task.Task.__init__
+        """
+        ImageMapperSubtask.__init__(self, *args, **kwargs)
+
+    def run(self, subExp, expandedSubExp, fullBBox, **kwargs):
+        """! Add `addAmount` to given subExposure.
+
+        @param[in] subExp the sub-exposure of `exposure` upon which to operate
+        @param[in] expandedSubExp the expanded sub-exposure of `exposure` upon which to operate
+        @param[in] fullBBox the bounding box of the original exposure
+        @return a `afw.Exp` or `afw.Exposure`
+        """
+        subExp = subExp.clone()
+        img = subExp.getMaskedImage()
+        img += self.config.addAmount
+        return subExp
+
+
+class TestImageMapReduceConfig(ImageMapReduceConfig):
+    """!
+    \anchor TestImageMapReduceConfig_
+
+    \brief Configuration parameters for the TestImageMapReduceTask
+    """
+    mapperSubtask = pexConfig.ConfigurableField(
+        doc="Mapper subtask to run on each subimage",
+        target=TestImageMapperSubtask,
+    )
+
+
+class TestImageMapperSubtask2(ImageMapperSubtask):
+    ConfigClass = TestImageMapperSubtaskConfig
+    _DefaultName = "ip_diffim_TestImageMapperSubtask2"
+
+    def __init__(self, *args, **kwargs):
+        """! Create the image gridding subTask
+        @param *args arguments to be passed to lsst.pipe.base.task.Task.__init__
+        @param **kwargs keyword arguments to be passed to lsst.pipe.base.task.Task.__init__
+        """
+        ImageMapperSubtask.__init__(self, *args, **kwargs)
+
+    def run(self, subExp, expandedSubExp, fullBBox, **kwargs):
+        """!Return the mean of the given subExposure.
+
+        @param[in] subExp the sub-exposure of `exposure` upon which to operate
+        @param[in] expandedSubExp the expanded sub-exposure of `exposure` upon which to operate
+        @param[in] fullBBox the bounding box of the original exposure
+        @return a float, the mean of `subExp.getMaskedImage().getArray()`.
+        """
+        return subExp.getMaskedImage().getImage().getArray().mean()
+
+
+class TestImageMapReduceConfig2(ImageMapReduceConfig):
+    """!
+    \anchor TestImageMapReduceConfig_
+
+    \brief Configuration parameters for the TestImageMapReduceTask2
+    """
+    mapperSubtask = pexConfig.ConfigurableField(
+        doc="Mapper subtask to run on each subimage",
+        target=TestImageMapperSubtask2,
+    )
+
+
+class ImageMapReduceTest(lsst.utils.tests.TestCase):
+    """!A test case for the image gridded processing task
+    """
+
+    def setUp(self):
+        self.exposure = afwImage.ExposureF(128, 128)
+        self.exposure.setPsf(measAlg.DoubleGaussianPsf(11, 11, 2.0, 3.7))
+        mi = self.exposure.getMaskedImage()
+        mi[:, :] = 0.
```

Comment:
Huh. I just looked, and didn't see an Exposure constructor that would assign a value to the entire maskedImage. Could you ask slack#dm if such a constructor exists, and if not file a ticket against afw?